### PR TITLE
Adding ability to inject dependencies into terminal delegates

### PR DESCRIPTION
**Pull request by srogovtsev**
* see autofac/Autofac.Owin#6

Sorry it took so long.

### CODE REVIEW

File: src/Autofac.Integration.Owin/AutofacAppBuilderRunExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        /// </code>
+        /// </example>
+        /// <seealso cref="AutofacAppBuilderExtensions.UseAutofacLifetimeScopeInjector(IAppBuilder, ILifetimeScope)"/>
+        public static void RunFromContainer<T>(this IAppBuilder app, Func<T,IOwinContext,Task> handler)
+        {
+            if (app == null)
+            {
+                throw new ArgumentNullException("app");
+            }
+
```
Comment (tillig) [inline src/Autofac.Integration.Owin/AutofacAppBuilderRunExtensions.cs]: Need to check handler for null.

**Comment by tillig** [comment]
Really nice. If you can add the test for null handler I'll pull it into develop. Thanks for the great contribution!

File: src/Autofac.Integration.Owin/AutofacAppBuilderRunExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        /// <param name="app">The application builder.</param>
+        /// <param name="handler">The handler to invoke. The constructed middleware will be available as a parameter.</param>
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown if <paramref name="app"/> is <see langword="null"/>.
+        /// </exception>
+        /// <example>
+        /// <code lang="C#">
+        /// app
+        ///   .UseAutofacLifetimeScopeInjector(container)
+        ///   .Run&lt;Uploader&gt;((uploader, owinContext) =&gt; uploader.InvokeAsync(owinContext));
```
Comment (tillig) [inline src/Autofac.Integration.Owin/AutofacAppBuilderRunExtensions.cs]: `RunFromContainer` rather than `Run` in the sample here.

File: src/Autofac.Integration.Owin/AutofacAppBuilderRunExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static class AutofacAppBuilderRunExtensions
+    {
+        /// <summary>
+        /// Adds a middleware to the OWIN pipeline that will be constructed using Autofac and that does not have next middleware
+        /// </summary>
+        /// <param name="app">The application builder.</param>
+        /// <param name="handler">The handler to invoke. The constructed middleware will be available as a parameter.</param>
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown if <paramref name="app"/> is <see langword="null"/>.
```
Comment (tillig) [inline src/Autofac.Integration.Owin/AutofacAppBuilderRunExtensions.cs]: `Thrown if <paramref name="app"/> or <paramref name="handler"/> is <see langword="null"/>.`

**Comment by srogovtsev** [comment]
Here you have it (not sure whether GitHub notified you of my commits)

**Comment by tillig** [comment]
It actually didn't notify me, which is weird.

Anyway, pulled in, and thanks again for the great work!

**Comment by alexmg** [comment]
Thanks for the contribution @srogovtsev.

**Comment by srogovtsev** [comment]
It was nice contributing - at last - to something that I use for my own development. Giving, not taking.
